enterprise/coderd/prebuilds/metricscollector_test.go

```diff
@@ -1,73 +1,247 @@
+			shouldIncrementPrebuildsCreated: ptr.To(1.0),
+			shouldSetDesiredPrebuilds:       ptr.To(1.0),
+			shouldSetEligiblePrebuilds:      ptr.To(0.0),
+		},
+		{
+			name:                            "prebuild running",
+			transitions:                     []database.WorkspaceTransition{database.WorkspaceTransitionStart},
+			jobStatuses:                     []database.ProvisionerJobStatus{database.ProvisionerJobStatusSucceeded},
+			initiatorIDs:                    []uuid.UUID{prebuilds.OwnerID},
+			ownerIDs:                        []uuid.UUID{prebuilds.OwnerID},
+			shouldIncrementPrebuildsCreated: ptr.To(1.0),
+			shouldSetDesiredPrebuilds:       ptr.To(1.0),
+			shouldSetActualPrebuilds:        ptr.To(1.0),
+			shouldSetEligiblePrebuilds:      ptr.To(0.0),
+		},
+		{
+			name:                            "prebuild failed",
+			transitions:                     allTransitions,
+			jobStatuses:                     []database.ProvisionerJobStatus{database.ProvisionerJobStatusFailed},
+			initiatorIDs:                    []uuid.UUID{prebuilds.OwnerID},
+			ownerIDs:                        []uuid.UUID{prebuilds.OwnerID, uuid.New()},
+			shouldIncrementPrebuildsCreated: ptr.To(1.0),
+			shouldIncrementPrebuildsFailed:  ptr.To(1.0),
+			shouldSetDesiredPrebuilds:       ptr.To(1.0),
+			shouldSetActualPrebuilds:        ptr.To(0.0),
+			shouldSetEligiblePrebuilds:      ptr.To(0.0),
+		},
+		{
+			name:                             "prebuild assigned",
+			transitions:                      allTransitions,
+			jobStatuses:                      allJobStatuses,
+			initiatorIDs:                     []uuid.UUID{prebuilds.OwnerID},
+			ownerIDs:                         []uuid.UUID{uuid.New()},
+			shouldIncrementPrebuildsCreated:  ptr.To(1.0),
+			shouldIncrementPrebuildsAssigned: ptr.To(1.0),
+			shouldSetDesiredPrebuilds:        ptr.To(1.0),
+			shouldSetActualPrebuilds:         ptr.To(0.0),
+			shouldSetEligiblePrebuilds:       ptr.To(0.0),
+		},
+		{
+			name:                             "workspaces that were not created by the prebuilds user are not counted",
+			transitions:                      allTransitions,
+			jobStatuses:                      allJobStatuses,
+			initiatorIDs:                     []uuid.UUID{uuid.New()},
+			ownerIDs:                         []uuid.UUID{uuid.New()},
+			shouldIncrementPrebuildsCreated:  nil,
+			shouldIncrementPrebuildsFailed:   nil,
+			shouldIncrementPrebuildsAssigned: nil,
+			shouldSetDesiredPrebuilds:        ptr.To(1.0),
+			shouldSetActualPrebuilds:         ptr.To(0.0),
+			shouldSetEligiblePrebuilds:       ptr.To(0.0),
+		},
+	}
+	for _, test := range tests {
+		for _, transition := range test.transitions {
+			for _, jobStatus := range test.jobStatuses {
+				for _, initiatorID := range test.initiatorIDs {
+					for _, ownerID := range test.ownerIDs {
+						t.Run(fmt.Sprintf("transition:%s/jobStatus:%s", transition, jobStatus), func(t *testing.T) {
```

Comment:
oh man.. what a complexity :D 

Perhaps take a look at `insights-metrics.json`. You can define test results as .json files, and have them unmarshalled and compared.

Reply:
what specifically is the complexity that you are referring to?